FAQ.md

```diff
@@ -98,8 +102,89 @@ This also means that if a class has a private field named `x`, code outside the
 
 ## Why is encapsulation a goal of this proposal?
 
-1. There are already good ways to provide "hidden" but not encapsulated fields, like Symbols as property names.
-
 1. Library authors have found that their users will start to depend on any exposed part of their interface, even undocumented parts. They do not generally consider themselves free to break their user's pages and applications just because those users were depending upon some part of the library's interface which the library author did not intend them to depend upon. As a consequence, they would like to have hard private state to be able to hide implementation details in a more complete way.
 
+1. While it's already possible to *model* true encapsulation using either per-instance closures or WeakMaps (see below), both approaches are unergonomic to use with classes and have memory usage semantics which may be surprising. Furthermore, per-instance closures don't allow instances of the same class to access each other's private fields (see [above](#why-does-this-proposal-allow-accessing-private-fields-of-other-instances-of-the-same-class-dont-other-languages-normally-forbid-that)), and with WeakMaps there's a non-obvious risk of exposing private data. WeakMaps are also likely to be slower than private fields could be.
+
+1. "Hidden" but not encapsulated properties are likewise already possible through the use of Symbols as property names (see below).
+
+Consensus in the committee is that we'd like a more ergonomic way to provide true encapsulation. As such, this proposal is currently moving forward with hard-private fields, letting [decorators](https://github.com/tc39/proposal-private-fields/blob/master/DECORATORS.md) or other mechanisms provide classes an opt-in escape hatch. There's still some debate about whether an escape hatch should exist by default, with some means to opt-out.
+
 See [this issue](https://github.com/tc39/proposal-private-fields/issues/33) for more.
+
+### How can you model encapsulation using WeakMaps?
+
+```js
+const Person = (function(){
+  const privates = new WeakMap;
+  let ids = 0;
+  return class Person {
+    constructor(name) {
+      this.name = name;
+      privates.set(this, {id: ids++});
+    }
+    equals(otherPerson) {
+      return privates.get(this).id === privates.get(otherPerson).id;
+    }
+  }
+})();
+let alice = new Person('Alice');
+let bob = new Person('Bob');
+alice.equals(bob); // false
+```
+
+There's a potential pitfall with this approach, though. Suppose we want to add a custom callback to construct greetings:
+
+```js
+const Person = (function(){
+  const privates = new WeakMap;
```

Comment:
Worth mentioning that the WeakMap approach fails cross-realm, but true hard-private state would not?